docs/component-specs.md

```diff
@@ -0,0 +1,29 @@
+# Spec files for agent-managed components
+
+Spec files are YML files that describe the capabilities of a component and when / how to run it. They are used by Agent to convert a user-provided policy into a configured running process.
+
+[This documentation should be expanded](https://github.com/elastic/elastic-agent/issues/2690)
+
+## Preventions
+
+Inputs may include a `runtime.preventions` section containing [EQL conditions](https://www.elastic.co/guide/en/elasticsearch/reference/current/eql-syntax.html#eql-syntax-conditions) which should prevent the use of that input if any are true. Each prevention should include a `condition` in EQL syntax and a `message` that will be displayed if that condition prevents the use of this input.
```

Comment:
The previous section talks about "components" while here we talk about "inputs".  It might be good to be consistent or provide a relationship between the two concepts.

Reply:
Thanks, "components" is the right term, "inputs" is just easy to slip into since all components are inputs except for one 😅